server/src/main/java/io/druid/server/QueryResource.java

```
@@ -183,6 +183,7 @@ public void write(OutputStream outputStream) throws IOException, WebApplicationE
                   {
                     // json serializer will always close the yielder
                     jsonWriter.writeValue(outputStream, yielder);
+                    outputStream.flush(); // Some types of OutputStream suppress flush errors in the .close() method.
```

Try with resources swallows exceptions in flush.